sdk/dotnet/Pulumi/Core/InputList.cs

```diff
@@ -57,11 +57,21 @@ private InputList(Output<ImmutableArray<T>> values)
 
         public void Add(params Input<T>[] inputs)
         {
-            // Make an Output from the values passed in, mix in with our own Output, and combine
-            // both to produce the final array that we will now point at.
-            _outputValue = Output.Concat(_outputValue, Output.All(inputs));
+            _outputValue = Concat(inputs);
         }
 
+        public void Add(InputList<T> inputs)
+        {
+            _outputValue = Concat(inputs);
+        }
+
+        /// <summary>
+        /// Concatenates the values in this list with the value in <paramref name="other"/>,
+        /// returning the concatenated sequence in a new <see cref="InputList{T}"/>.
+        /// </summary>
+        public InputList<T> Concat(Input<T> other)
```

Comment:
I'd normally (and at least in dotnet F# it seems this way) use Concat for two sequence parameters, and Add/Append for a sequence and a single value.

Reply:
Happy to rename things if that helps.

Reply:
This can actually be removed with the `params` signature restored since it uses the other overload.

Reply:
(Which highlights the mismatch from using `params` since it's called multiple times by the collection initializer, converted to an `InputList` from an array of _one_ item, and then passed to `Concat`.)